File: app_canibuythat/src/main/java/com/gb/canibuythat/model/Transaction.java
package com.gb.canibuythat.model;

import org.threeten.bp.ZonedDateTime;

import lombok.Builder;
import lombok.Getter;

@Builder
@Getter
public class Transaction {
    private float amount;
    private ZonedDateTime created;
    private String currency;
    private String description;
    private String id;
    private String merchant;
    private String notes;
    private boolean isLoad;
    private ZonedDateTime settled;
    private String category;
}

File: app_canibuythat/src/main/java/com/gb/canibuythat/model/Transaction.kt
package com.gb.canibuythat.model

import com.gb.canibuythat.db.model.ApiSpending
import java.time.ZonedDateTime

class Transaction(val amount: Int,
                  val created: ZonedDateTime,
                  val description: String?,
                  val id: String,
                  val category: ApiSpending.Category) {

    override fun toString(): String {
        return "Transaction(category='$category', amount=$amount, created=$created, description='$description', id='$id')"
    }
}